errortypes.go

```diff
@@ -4,379 +4,460 @@
 package errors
 
 import (
+	stderror "errors"
 	"fmt"
+	"strings"
 )
 
-// wrap is a helper to construct an *wrapper.
-func wrap(err error, format, suffix string, args ...interface{}) Err {
-	newErr := Err{
-		message:  fmt.Sprintf(format+suffix, args...),
-		previous: err,
+// a ConstError is a prototype for a certain type of error
+type ConstError string
```

Comment:
Why is ConstError exported if none of the wrapping utilities are exported?

Reply:
fair point, I think we could make it `constError`. @tlm?

Reply:
Though we are exporting constants like `Timeout`, `NotFound` etc which have type `ConstError`, so does that justify it?

Reply:
I think we should export it. It's a useful type that provides a utility if other things want to mint quick errors.

A utility library like this should export its utilities.

The reason we don't export the wrap stuff as it's purely there to keep the business logic the same for this re-work